pytorch_blade/src/compiler/backends/engine_class.cpp

```diff
@@ -160,82 +160,88 @@ c10::intrusive_ptr<EngineClass> EngineClass::Deserialize(
-          // The following four lines expose methods of the
-          // MyStackClass<std::string> class as-is. `torch::class_` will
-          // automatically examine the argument and return types of the
-          // passed-in method pointers and expose these to Python and
-          // TorchScript accordingly. Finally, notice that we must take the
-          // *address* of the fully-qualified method name, i.e. use the unary
-          // `&` operator, due to C++ typing rules.
-          .def("dump_attr_to_file", &EngineClass::DumpAttrToFile)
-          .def("dump_model_proto", &EngineClass::DumpModelProto)
-          .def("get_attr_string", &EngineClass::GetAttrString)
-          .def("get_attr_keys", &EngineClass::GetAttrKeys)
-          .def("last_inputs", &EngineClass::last_inputs)
-          .def("last_outputs", &EngineClass::last_outputs)
-          // class_<>::def_pickle allows you to define the serialization
-          // and deserialization methods for your C++ class.
-          // Currently, we only support passing stateless lambda functions
-          // as arguments to def_pickle
-          .def_pickle(
-              // __getstate__
-              // This function defines what data structure should be produced
-              // when we serialize an instance of this class. The function
-              // must take a single `self` argument, which is an intrusive_ptr
-              // to the instance of the object. The function can return
-              // any type that is supported as a return value of the TorchScript
-              // custom operator API. In this instance, we've chosen to return
-              // a std::vector<std::string> as the salient data to preserve
-              // from the class.
-              [](const c10::intrusive_ptr<EngineClass>& self)
-                  -> EngineClass::SerialType { return self->Serialize(); },
-              // __setstate__
-              // This function defines how to create a new instance of the C++
-              // class when we are deserializing. The function must take a
-              // single argument of the same type as the return value of
-              // `__getstate__`. The function must return an intrusive_ptr
-              // to a new instance of the C++ class, initialized however
-              // you would like given the serialized state.
-              [](EngineClass::SerialType serialized)
-                  -> c10::intrusive_ptr<EngineClass> {
-                // A convenient way to instantiate an object and get an
-                // intrusive_ptr to it is via `make_intrusive`. We use
-                // that here to allocate an instance of
-                // MyStackClass<std::string> and call the single-argument
-                // std::vector<std::string> constructor with the serialized
-                // state.
-                return EngineClass::Deserialize(std::move(serialized));
-              });
+  static std::once_flag flag;
```

Comment:
Does this function still have to be exposed to pybind if it called by global initializer...

Reply:
Already removed.